internal/functions/coviddata/download-and-count-vaccinations.go

```diff
@@ -0,0 +1,118 @@
+package coviddata
+
+import (
+	"context"
+	"encoding/json"
+	"github.com/covid19cz/erouska-backend/internal/constants"
+	"github.com/covid19cz/erouska-backend/internal/logging"
+	"github.com/covid19cz/erouska-backend/internal/store"
+	httputils "github.com/covid19cz/erouska-backend/internal/utils/http"
+	"github.com/sethvargo/go-envconfig"
+	"io/ioutil"
+	"net/http"
+	"time"
+)
+
+func fetchVaccinationsData(client HTTPClient) (*VaccinationsAggregatedData, error) {
+
+	var ctx = context.Background()
+	logger := logging.FromContext(ctx)
+
+	var vaccinationMetricsConfig vaccinationMetricsConfig
+	if err := envconfig.Process(ctx, &vaccinationMetricsConfig); err != nil {
+		logger.Debugf("Could not load vaccinationMetricsConfig: %v", err)
+		return nil, err
+	}
+
+	req, err := http.NewRequest(http.MethodGet, vaccinationMetricsConfig.URL, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	res, getErr := client.Do(req)
+	if getErr != nil {
+		return nil, err
+	}
+
+	if res.Body != nil {
+		defer res.Body.Close()
+	}
+
+	body, readErr := ioutil.ReadAll(res.Body)
+	if readErr != nil {
+		return nil, err
+	}
+
+	var r vaccinationDownloadRequest
+
+	jsonErr := json.Unmarshal(body, &r)
+	if jsonErr != nil {
+		return nil, err
+	}
+
+	data := r.Data
+
+	sumByDate := make(map[string]VaccinationsAggregatedData)
+	totalFirstDose := 0
+	totalSecondDose := 0
+	modification, _ := time.Parse(time.RFC3339, r.Modified)
+	lastDate := "" // date nearest today
+
+	// always must be processed all data in JSON due to possible fixes
+	for _, v := range data {
+		date := reformatDate(v.Date)
+		vd := sumByDate[date]
+		vd.Date = date
+		vd.DailyFirstDose += v.FirstDose
+		vd.DailySecondDose += v.SecondDose
+		totalFirstDose += v.FirstDose
+		totalSecondDose += v.SecondDose
+		sumByDate[date] = vd
+		if vd.Date > lastDate {
+			lastDate = vd.Date
+		}
+	}
+
+	lastDateVaccinations := VaccinationsAggregatedData{
+		Date:            lastDate,
+		Modified:        modification.Unix(),
+		DailyFirstDose:  sumByDate[lastDate].DailyFirstDose,
+		DailySecondDose: sumByDate[lastDate].DailySecondDose,
+		TotalFirstDose:  totalFirstDose,
+		TotalSecondDose: totalSecondDose,
+	}
+
+	logger.Debugf("Fetched new vaccinations metrics: %+v", lastDateVaccinations)
+
+	return &lastDateVaccinations, nil
+}
+
+// DownloadAndCountVaccinations downloads vaccinations metrics json and writes it to firestore
+func DownloadAndCountVaccinations(w http.ResponseWriter, r *http.Request) {
+	var ctx = context.Background()
+	logger := logging.FromContext(ctx)
+	client := store.Client{}
```

Comment:
Please
* Move this function to the top of the file
* Use named loggers :-)
* Use this public function only to create a configuration that is later passed to another function that does the job (as is in most of the other functions, look at `EfgsUploadKeys` for example